ddpui/core/git_manager.py

```
@@ -661,3 +676,116 @@ def check_remote_repository_empty_static(remote_url: str, pat: str = None) -> bo
+            # Try to create fine-grained PAT
+            pat_data = GitManager._github_api_request(
+                url="https://api.github.com/user/tokens",
+                pat=org_admin_pat,
+                payload=payload,
+                method="POST",
+            )
+
+            logger.info(f"Created fine-grained PAT for repository {repo_name}")
+            return pat_data["token"]
+
+        except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)